compiler/K_lift.fx

```diff
@@ -597,14 +625,23 @@ fun lift_all(kmods: kmodule_t list)
         val e =
         match e {
         | KDefFun kf =>
-            val {kf_name, kf_params, kf_body, kf_closure, kf_loc} = *kf
+            val {kf_name, kf_loc} = *kf
+            val is_declosured = declosured_functions.mem(kf_name)
+            if(is_declosured) {
+                val {kf_params} = *kf
+                val (kcv_freevars, _) = get_closure_freevars(kf_name, kf_loc)
+                val params_addition = [: for (fv,_) <- kcv_freevars { fv } :]
```

Comment:
I think, this will only work with immutable free variables. If there is a mutable free variable, we need to pass the reference to the function, not the original freevar. Since there is no type checking in K-form, the concatenation step could possibly be postponed till the last stage of lambda lifting, where for each free var we have its normal substitute and the "reference" substitute. And we should add them into the list of format parameters, not the original freevars, which will become actual arguments, not the formal parameters.